### PR TITLE
[FEATURE] Nucleotide16

### PR DESCRIPTION
**Pull request by sarahet**

### CODE REVIEW

File: include/seqan3/alphabet/nucleotide/nucl16_sequence.hpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#pragma once
```
Comment (h-2) [inline include/seqan3/alphabet/nucleotide/nucl16_sequence.hpp]: copyright header

File: include/seqan3/alphabet/nucleotide/nucl16_sequence.hpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+{
+
+inline nucl16_vector operator "" _nucl16(const char * s, std::size_t n)
+{
+    nucl16_vector r;
+    r.resize(n);
+
+    std::transform(s, s + n, r.begin(), [] (const char & c)
+    {
+        return nucl16{nucl16::char_to_value[c]};
```
Comment (h-2) [inline include/seqan3/alphabet/nucleotide/nucl16_sequence.hpp]: use `from_char` instead

File: include/seqan3/alphabet/nucleotide/nucl16_sequence.hpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+inline nucl16_string operator "" _nucl16s(const char * s, std::size_t n)
+{
+    nucl16_string r;
+    r.resize(n);
+
+    std::transform(s, s + n, r.begin(), [] (const char & c)
+    {
+        return nucl16{nucl16::char_to_value[c]};
```
Comment (h-2) [inline include/seqan3/alphabet/nucleotide/nucl16_sequence.hpp]: use `from_char` instead

File: include/seqan3/alphabet/nucleotide/nucl16.hpp [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+    {
+        value = c;
+    }
+    constexpr operator c_type() const
+    {
+        return value;
+    }
+
+    // explicit compatibility to char
+    explicit constexpr operator char() const
```
Comment (cpockrandt) [inline include/seqan3/alphabet/nucleotide/nucl16.hpp]: replace char() with char_type()

**Pull request merged**